docs/source/reference/migration/from-router-v1.mdx

```diff
@@ -411,7 +411,18 @@ For more information about [configuring traffic shaping](../../routing/performan
 
 If, in your Rust plugin, you were using either the subgraph::Request::subgraph_name or the subgraph::Response::subgraph_name, please note that these are no longer `Option`, but are mandatory.
 
-<!-- PR: https://github.com/apollographql/router/pull/6486 -->
+### Emitting metrics in Rust plugins
+
+Previously, Rust plugins could use the Router's internal metrics system using the `tracing` macros. This is no longer supported, and the opentelemetry crates should be used instead.
+
+`tracing` field names starting with these strings are no longer treated specially:
+- `counter.`
+- `histogram.`
+- `monotonic_counter.`
+- `value.`
+
+Instead, use the [OpenTelemetry](https://docs.rs/opentelemetry/latest/opentelemetry/) crates.
+You can use the new `apollo_router::metrics::meter_provider()` API to access the router's global meter provider to register your instruments.
```

Comment:
I think you have renamed this to meter_provider_internal() unless this is referencing something else...

Reply:
I renamed the existing `meter_provider()` function to `meter_provider_internal()`, because it returns a non-public type with custom feature extensions.

There is also a new `fn meter_provider()` that returns an `impl MeterProvider`, so we *only* expose the API of the meter provider type from the `opentelemetry` crate, and not our own implementation details.